azurerm/resource_arm_batch_pool.go

```diff
@@ -0,0 +1,476 @@
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"offer": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"sku": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"version": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ForceNew: true,
+						},
+					},
+				},
+				Set: resourceArmVirtualMachineStorageImageReferenceHash,
+			},
+			"node_agent_sku_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"stop_pending_resize_operation": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceArmBatchPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchPoolClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure Batch pool creation.")
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	accountName := d.Get("account_name").(string)
+	name := d.Get("name").(string)
+	displayName := d.Get("display_name").(string)
+	vmSize := d.Get("vm_size").(string)
+
+	parameters := batch.Pool{
+		PoolProperties: &batch.PoolProperties{
+			VMSize:      &vmSize,
+			DisplayName: &displayName,
+		},
+	}
+
+	scaleMode := azure.BatchPoolScaleMode(d.Get("scale_mode").(string))
+
+	if scaleMode == azure.BatchPoolAutoScale {
+		autoScaleEvaluationInterval := d.Get("autoscale_evaluation_interval").(string)
+		autoScaleFormula := d.Get("autoscale_formula").(string)
+
+		if autoScaleFormula == "" {
+			return fmt.Errorf("Error: when scale mode is set to Auto, auto_scale formula cannot be empty")
+		}
+
+		parameters.PoolProperties.ScaleSettings = &batch.ScaleSettings{
+			AutoScale: &batch.AutoScaleSettings{
+				EvaluationInterval: &autoScaleEvaluationInterval,
+				Formula:            &autoScaleFormula,
+			},
+		}
+	} else if scaleMode == azure.BatchPoolFixedScale {
+		targetDedicatedNodes := int32(d.Get("target_dedicated_nodes").(int))
+		targetLowPriorityNodes := int32(d.Get("target_low_priority_nodes").(int))
+		resizeTimeout := d.Get("resize_timeout").(string)
+
+		parameters.PoolProperties.ScaleSettings = &batch.ScaleSettings{
+			FixedScale: &batch.FixedScaleSettings{
+				ResizeTimeout:          &resizeTimeout,
+				TargetDedicatedNodes:   &targetDedicatedNodes,
+				TargetLowPriorityNodes: &targetLowPriorityNodes,
+			},
+		}
+	} else {
+		return fmt.Errorf("Error: scale mode should be either AutoScale of FixedScale")
+	}
+
+	nodeAgentSkuID := d.Get("node_agent_sku_id").(string)
+	storageImageReferenceSet := d.Get("storage_image_reference").(*schema.Set)
+
+	if storageImageReferenceSet == nil || storageImageReferenceSet.Len() == 0 {
+		return fmt.Errorf("Error: storage image reference should be defined")
+	}
+
+	storageImageRef := storageImageReferenceSet.List()[0].(map[string]interface{})
```

Comment:
Could we separate this out into an `expand` function?

Reply:
Done.